spec/lib/rev/models/order_request_spec.rb

```diff
@@ -41,19 +46,57 @@
 
   describe 'TranscriptionOptions' do
     it 'is InputOptions' do
-      options = Rev::TranscriptionOptions.new([{}], {})
+      inputs = create_input()
+      options = Rev::TranscriptionOptions.new(inputs, {})
       options.must_be_kind_of Rev::InputOptions
     end
+
+    it 'rejects glossary of invalid size' do
+      oversize_glossary = []
+      for x in 0..GLOSSARY_ENTRIES_LIMIT_TEST do
+        oversize_glossary << 'testing'
+      end
+      inputs = create_input(glossary: oversize_glossary)
+      proc { Rev::TranscriptionOptions.new(inputs) }.must_raise ArgumentError
+    end
+
+    it 'rejects glossary with invalid terms' do
+      oversize_glossary_term = "A" * (GLOSSARY_ENTRY_LENGTH_LIMIT_TEST + 1)
+      inputs = create_input(glossary: [oversize_glossary_term])
+      proc { Rev::TranscriptionOptions.new(inputs) }.must_raise ArgumentError
+    end
+
+    it 'rejects speaker names of invalid size' do
+      oversize_speaker_names = []
+      for x in 0..SPEAKER_ENTRIES_LIMIT_TEST do
+        oversize_speaker_names << 'testing'
+      end
+      inputs = create_input(speaker_names: oversize_speaker_names)
+      proc { Rev::TranscriptionOptions.new(inputs) }.must_raise ArgumentError
+    end
+
+    it 'rejects speaker names if name is too long' do
+      oversize_speaker_name = "A" * (SPEAKER_ENTRY_LENGTH_LIMIT_TEST + 1)
+      inputs = create_input(speaker_names: [oversize_speaker_name])
+      proc { Rev::TranscriptionOptions.new(inputs) }.must_raise ArgumentError
+    end
+
+    it 'rejects invalid accents' do
+      inputs = create_input(accents: ['invalid'])
+      proc { Rev::TranscriptionOptions.new(inputs) }.must_raise ArgumentError
+    end
   end
 
   describe 'CaptionOptions' do
     it 'is InputOptions' do
-      options = Rev::CaptionOptions.new([{}], {})
+      inputs = create_input()
+      options = Rev::CaptionOptions.new(inputs, {})
```

Comment:
Old tests were passing empty hash's instead of objects like they should have been.